onnxruntime/contrib_ops/cuda/layer_norm_impl.cu

```diff
@@ -316,21 +324,20 @@ __global__ void cuApplyLayerNorm(
     SharedMemory<U> shared;
     U* buf = shared.getPointer();
     U mu, sigma2;
-    cuWelfordMuSigma2(vals, n1, n2, i1, mu, sigma2, buf);
+    cuWelfordMuSigma2<T, U, simplified>(vals, n1, n2, i1, mu, sigma2, buf);
     const T* lvals = vals + i1 * n2;
     T* ovals = output_vals + i1 * n2;
     U c_invvar = rsqrt(sigma2 + epsilon);
     const int numx = blockDim.x * blockDim.y;
     const int thrx = threadIdx.x + threadIdx.y * blockDim.x;
-    if (gamma != NULL && beta != NULL) {
-      for (int i = thrx; i < n2; i += numx) {
-        U curr = static_cast<U>(lvals[i]);
-        ovals[i] = gamma[i] * static_cast<T>(c_invvar * (curr - mu)) + beta[i];
-      }
-    } else {
-      for (int i = thrx; i < n2; i += numx) {
-        U curr = static_cast<U>(lvals[i]);
-        ovals[i] = static_cast<T>(c_invvar * (curr - mu));
+    for (int i = thrx; i < n2; i += numx) {
+      U curr = static_cast<U>(lvals[i]);
+      T gamma_i = (gamma != NULL) ? gamma[i]: (T)0;
```

Comment:
the default value of gamma should be 1. Consider if gamma is null, the computation below will result in output of 0 which whereas it should be:
 `ovals[i] = static_cast<T>(c_invvar * (curr - mu));`

Reply:
You're right. The default should be 1. Fixed that